units/provisioning_create_host.go

```diff
@@ -163,6 +177,42 @@ func (j *createHostJob) Run(ctx context.Context) {
 	j.AddError(j.createHost(ctx))
 }
 
+func (j *createHostJob) selfThrottle() bool {
+	var (
+		numProv            int
+		runningHosts       int
+		distroRunningHosts int
+		err                error
+	)
+
+	numProv, err = host.CountStartedTaskHosts()
```

Comment:
I have seen several cases recently where a bunch of hosts are not in the running state but are running tasks. Since we've allowed hosts to run tasks while still in starting, I think the risk is that these hosts will cause provisioning to stop. We would need to address that problem first before this is safe to deploy.

Reply:
This counts everything that has started but hasn't completed provisioning (just building and starting,) I think hosts can't start running tasks till they make it to provisioning. 

At any rate, the throttle exists to reduce throughput ("starvation") but with the effect of improving quality of service across the board, and hopefully by not running headlong into the limits of our AWS account, we'll end up being able to start more hosts with less friction, but we have to throttle somewhere.